File: st-exerciseviewer/src/main/java/de/saring/exerciseviewer/parser/ExerciseParserInfo.java
package de.saring.exerciseviewer.parser;

/**
 * This class contains informations of an ExerciseParser implementation.
 * 
 * @author  Stefan Saring
 * @version 1.0
 */
public class ExerciseParserInfo {

    /** The name of the parser. */
    private String name;
    
    /** List of exercise file suffixes which can be read by this parser. */
    private String[] suffixes;

    /**
     * Creates a new ExerciseParserInfo instance.
     * @param name list of exercise file suffixes which can be read by this parser
     * @param suffixes list of exercise file suffixes which can be read by this parser
     */
    public ExerciseParserInfo (String name, String[] suffixes) {
        this.name = name;
        this.suffixes = suffixes;
    }

    /***** BEGIN: Generated Getters and Setters *****/
    
    public String getName () {
        return name;
    }

    public String[] getSuffixes () {
        return suffixes;
    }

    /***** END: Generated Getters and Setters *****/
}

File: st-exerciseviewer/src/main/kotlin/de/saring/exerciseviewer/parser/ExerciseParserInfo.kt
package de.saring.exerciseviewer.parser

/**
 * This class contains the name and the supported files of an ExerciseParser.
 *
 * @property name The name of the parser.
 * @property suffixes List of exercise file suffixes which can be read by this parser.
 *
 * @author Stefan Saring
 */
class ExerciseParserInfo(

        val name: String,
        val suffixes: List<String>)